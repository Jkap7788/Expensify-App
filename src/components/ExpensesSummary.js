import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = (props) => (
    <div>
        <p>
            Viewing {props.expensesCount}
            {props.expensesCount === 1 ? <span> expense </span> : <span> expenses </span>}
            totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
        </p>

    </div>

)

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expensesCount: expenses.length,
        expensesTotal: selectExpensesTotal(expenses)
    }
}

export default connect(mapStateToProps, undefined)(ExpensesSummary)