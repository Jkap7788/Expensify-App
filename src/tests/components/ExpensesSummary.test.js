import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('Should render ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={50.2} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render ExpensesSummary correctly with multiple or zero expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={150.75} />)
    expect(wrapper).toMatchSnapshot()
})