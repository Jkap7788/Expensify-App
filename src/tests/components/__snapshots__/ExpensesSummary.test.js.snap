// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
    <span>
       expense 
    </span>
    totalling 
    $0.50
  </p>
</div>
`;

exports[`Should render ExpensesSummary correctly with multiple or zero expenses 1`] = `
<div>
  <p>
    Viewing 
    3
    <span>
       expenses 
    </span>
    totalling 
    $1.51
  </p>
</div>
`;
